### PR TITLE
🔀✨ NWK bearbeiten

### PR DESCRIPTION
**Pull request by LuisEinig**
**Description:**  

Added the NWK bearbeiten feature. Also moved NWK components into a folder.

**Reference:**

Issue: #17 

Notifying additional team members:

@mirrodi @MrSebastian @AnHo314

---
**Definition of Done (DoD):**

- [x] Unittests gepflegt
- [x] Dokumentation ist gepflegt

### CODE REVIEW

File: praktikumsplaner-backend/src/main/java/de/muenchen/oss/praktikumsplaner/rest/NwkController.java [hunk 1]
```diff
@@ -49,4 +49,15 @@ public List<NwkDto> getNwks(@RequestParam(name = "status", required = false) Str
         }
         return nwkService.findAllNwks();
     }
+
+    @PreAuthorize("hasRole('ROLE_' + T(de.muenchen.oss.praktikumsplaner.security.AuthoritiesEnum).AUSBILDUNGSLEITUNG.name())")
+    @PutMapping
+    public void updateNwk(@RequestBody NwkDto nwkDto) {
+        if (!nwkService.NwkExistsById(nwkDto.id())) {
+            throw new ResponseStatusException(HttpStatus.BAD_REQUEST,
```
Comment (Tuckerle) [inline praktikumsplaner-backend/src/main/java/de/muenchen/oss/praktikumsplaner/rest/NwkController.java:57]: I think this should be a 404 Not Found, if we only allow modification and not creation via put we should tell the user that the resource he was trying to edit does not exist

File: praktikumsplaner-backend/src/main/java/de/muenchen/oss/praktikumsplaner/service/NwkService.java [hunk 2]
```diff
@@ -47,4 +47,12 @@ public List<NwkDto> findAllUnassignedNwksInCurrentMeldezeitraum() {
     public List<NwkDto> findAllNwks() {
         return nwkRepository.findAll().stream().map(nwkMapper::toDto).toList();
     }
+
+    public void saveNwk(final NwkDto nwkDto) {
+        nwkRepository.save(nwkMapper.toEntity(nwkDto));
+    }
+
+    public boolean NwkExistsById(final UUID id) {
```
Comment (Tuckerle) [inline praktikumsplaner-backend/src/main/java/de/muenchen/oss/praktikumsplaner/service/NwkService.java:55]: to camel case

File: praktikumsplaner-backend/src/main/java/de/muenchen/oss/praktikumsplaner/rest/NwkController.java [hunk 3]
```diff
@@ -49,4 +49,15 @@ public List<NwkDto> getNwks(@RequestParam(name = "status", required = false) Str
         }
         return nwkService.findAllNwks();
     }
+
+    @PreAuthorize("hasRole('ROLE_' + T(de.muenchen.oss.praktikumsplaner.security.AuthoritiesEnum).AUSBILDUNGSLEITUNG.name())")
+    @PutMapping
+    public void updateNwk(@RequestBody NwkDto nwkDto) {
```
Comment (Tuckerle) [inline praktikumsplaner-backend/src/main/java/de/muenchen/oss/praktikumsplaner/rest/NwkController.java:55]: should be final

**Review by Tuckerle — COMMENTED** [review]
Entering no Data for name, vorname or invalid data for Jahrgang does not prevent the the submission of the dialog but results in an error message

File: praktikumsplaner-frontend/frontend/src/composables/rules.ts [hunk 4]
```diff
@@ -40,13 +47,20 @@ export function useRules() {
             message;
     }
 
+    function regexRule(regex: RegExp, message = "error") {
+        return (value: string | null | undefined) =>
+            (value && regex.test(value)) || message;
+    }
+
```
Comment (Tuckerle) [inline praktikumsplaner-frontend/frontend/src/composables/rules.ts:54]: I think either this rule should be specific for the jahrgang or the email rule should be refactored to use this generic regexRule

Comment (LuisEinig) [reply praktikumsplaner-frontend/frontend/src/composables/rules.ts]: That's true. I removed the email rule, and replaced it with regex rule

**Review by Tuckerle — APPROVED** [review]

**Pull request merged**